From 1f94681ec6683efbae1dc2cab90b4baadd31d0df Mon Sep 17 00:00:00 2001
From: Tim Foster <tim.foster@joyent.com>
Date: Wed, 14 Aug 2019 22:38:46 +0100
Subject: [PATCH] OS-7921 SmartOS bash_completions should complete alias names
 (fix script) Reviewed by: Mike Gerdts <mike.gerdts@joyent.com> Reviewed by:
 John Levon <john.levon@joyent.com> Approved by: John Levon
 <john.levon@joyent.com>

---
 usr/src/cmd/nsadmin/bash/bash_completion.d/zone_alias | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/usr/src/cmd/nsadmin/bash/bash_completion.d/zone_alias b/usr/src/cmd/nsadmin/bash/bash_completion.d/zone_alias
index 75de543b08..b0a465060f 100644
--- a/usr/src/cmd/nsadmin/bash/bash_completion.d/zone_alias
+++ b/usr/src/cmd/nsadmin/bash/bash_completion.d/zone_alias
@@ -4,7 +4,7 @@ _zone_alias()
     alias="${COMP_WORDS[COMP_CWORD]}"
     RES=$(vmadm list -H -o alias,uuid | awk -v alias="$alias" '
         BEGIN { count = 0; uuid = ""; }
-        $1 ~ "^"alias" " { count = 1; aliases[1] = $1; uuid = $2; exit;}
+        $1 == alias { count = 1; aliases[1] = $1; uuid = $2; exit;}
         $1 ~ "^"alias { count += 1; aliases[count] = $1; uuid = $2; }
         END { if (count == 1) {
                   print "smartos_alias_completed " aliases[1] " " uuid;
-- 
2.21.0

